### ISSUE TITLE
The performance on the web page is different from the local one

### ISSUE THREAD

**Issue by demo-telo** [facebookresearch/segment-anything#587]
Dear developers, thank you for your great work. But I found that the effect of deploying the model (ViT-h) you provided locally is far inferior to the effect presented on the web page, whether it is from the fineness or the detection effect of some difficult images. Even the recently proposed new model HQ-SAM, which claims to be an improvement on SAM, is not as good as the web version.  The problem with this effect is not only in automatic mask generator mode(everything), but also in predictor mode(point/box). As you showed, there are actually not many parameters that can be adjusted in predictor mode. So have you been iterating on this model in the near future? Or is there any other way to post-process it?

**Comment by sunsunnyshine** [comment]
I'm confused too.

**Comment by cmayer** [comment]
Many thanks for this great tool. I am using it for science.

I have the same problem is the users above.  I uploaded an image to the web demo and it finds every object I want to find. If I use it locally, with the SamAutomaticMaskGenerator class and after trying very different parameters it always misses objects I expect to find. Is there a way to get the same performance locally is on the web interface. Maybe I am using the wrong detection class, the wrong parameters.

**Comment by cmayer** [comment]
I solved the problem for me. The default parameter for the number of points per side in the code is much lower than the parameter used in the web interface. Increasing this number from 32 to 80 or 100 gives much better results locally. The point grid is shown in the web interface and it is definitely much denser than 32x32 points.

**Comment by demo-telo** [comment]
> I solved the problem for me. The default parameter for the number of points per side in the code is much lower than the parameter used in the web interface. Increasing this number from 32 to 80 or 100 gives much better results locally. The point grid is shown in the web interface and it is definitely much denser than 32x32 points.

YES.I'm pretty sure SamAutomaticMaskGenerator can be better by changing some parameters. But there is almost nothing that the predictor can adjust, it just generates a mask based on my prompt points. Even if there is post-processing method, it cannot achieve a good effect. This phenomenon is especially obvious when the image size is large. I think this is due to Resizelongside. So if there is no adjustment of model parameters, I think there should be at least some pre-processing steps to achieve the effect of an online model. (For example, crop the image to the target area (3000 *3000-> 300 * 300) and then perform ViT encoder).

**Comment by zhywyt** [comment]
> I solved the problem for me. The default parameter for the number of points per side in the code is much lower than the parameter used in the web interface. Increasing this number from 32 to 80 or 100 gives much better results locally. The point grid is shown in the web interface and it is definitely much denser than 32x32 points.我为自己解决了这个问题。代码中每边点数的默认参数比web界面中使用的参数低得多。将这个数字从32增加到80或100会在局部产生更好的结果。点网格显示在Web界面中，它肯定比32x32点密集得多。

Congratulations! Could you share the code of yours? I would really appreciate it.
